### PR TITLE
Add changelog for release 0.6.0

### PR DESCRIPTION
**Pull request by sbastrakov**
Here is a draft. I started with a [helper script](https://github.com/ComputationalRadiationPhysics/picongpu/blob/dev/docs/propose_changelog.py), which is really nice. Then regrouped PRs and fixed their titles in the log where necessary. That took surprisingly long (or unsurprisingly for @psychocoderHPC i guess).

One thing i don't particularly like is that a lot of those lines are not actually relative to the old release as suggested. E.g. there was a bug introduced to `dev` after the release and later fixed in `dev`, so together they make no change relative to old release. But now i think it would be practically impossible to separate.

The first message before the changes is a bit drafty, but content-wise reflects what i think. The "thanks to" list is same as `creators` in #3923, same order as well.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,378 @@
 Changelog
 =========
 
+0.6.0
+-----
+
+**Date:** 2021-12-XX
```
Comment (sbastrakov) [inline CHANGELOG.md:7]: Needs to be fixed before releasing (here or in another PR)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,378 @@
...
+
+C++14, New Solvers, I/O via openPMD API, HIP Support
```
Comment (sbastrakov) [inline CHANGELOG.md:9]: I was doubting how to call those "solvers". Was considering "extended numerics" instead, but thought that was possibly confusing wrt floating-point arithmetic. Also, we internally call those things solvers, however that may be not obvious for everyone

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,378 @@
...
+
+This release switches to C++14 as minimum required version. Transition to C++17
+is planned for upcoming releases.
+
+We extended PIConGPU with a few new solvers. Binary collisions are now
+available. We added arbitrary-order FDTD Maxwell's solver. All field solvers are
+now compatible with perfectly matched layer absorber, which became default.
+There is a new way of generating incident field based on the total
```
Comment (sbastrakov) [inline CHANGELOG.md:17]: Maybe this should be reduced or excluded altogether? At the release, it's only for Yee (tho the laser is also only for Yee)

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,378 @@
...
+Franz Poeschel, Lennert Sprenger, Klaus Steiniger, Manhui Wang,
+Sebastian Starke, Maxence Thévenet, Richard Pausch, René Widera
+for contributions to this release!
+
+### Changes to "0.5.0"
+
+**User Input Changes:**
+ - Remove HDF5 I/O plugin (replaced with new openPMD plugin) #3361
+ - Remove ADIOS I/O plugin (replaced with new openPMD plugin) #3691
+ - Change default current solver to EmZ #3599
```
Comment (psychocoderHPC) [inline CHANGELOG.md:48]: We can remove this line, we are on the way to revert to Esirkepov :-( #3932

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,378 @@
...
+   - Add reference in LCT Example #3297
+   - Fix minor issues with the sphinx rst formating #3290
+   - Fix the name of probe fieldE and fieldB attribute in docs #3385
+   - Improve clarity of comments concerning density calculation and profile interface #3414
+   - Clarify comments for generating momentums based on temperature #3423
+   - Replace travis badges with gitlab ones #3451
+   - Update supported CUDA versions #3458
+   - Juwels profile update #3359
+   - Extend reminder to load environment in the docs with instructions for spack #3478
+   - Update license headers for 2021 #3486
```
Comment (psychocoderHPC) [inline CHANGELOG.md:281]: IMO we can remove this it is definitely not importent for the user

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,6 +1,378 @@
...
+   - Fix typo in FieldAbsorberTest #3528
```
Comment (psychocoderHPC) [inline CHANGELOG.md:282]: can be removed too

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,6 +1,378 @@
...
+   - Add fieldAbsorber runtime parameter to TBG_macros.cfg #3646
+   - Extend docs for the openPMD plugin regarding --openPMD.source #3667
+   - Update documentation of memory calculator #3669
+   - Summit template and documentation for asynchronous writing via ADIOS2 SST #3698
+   - Add DGX (A100) profile on Cori #3694
+   - Include clang-format in suggested contribution pipeline #3710
+   - Extend TBG_macros and help string for changing field absorber #3736
+   - Add link to conda picongpu-analysis-environment file to docs #3738
+   - Add a brief doc page on ways of adding a laser to a simulation #3739
+   - Update cupla to 0.3.0 release #3748
```
Comment (psychocoderHPC) [inline CHANGELOG.md:302]: can be removed

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,6 +1,378 @@
...
+   - add reference list #3273
+   - Fix docker documentation #3544
+   - Extend comments of CreateDensity #3837
+   - Refactor and document laser profiles #3798
+   - Mark CUDA 11.2 as supported version #3503
+   - Document intention of Pointer #3831
+   - Update ISAAC plugin documentation #3740
+   - Add a doxygen warning to not remove gamma filter in transition radiation #3857
+   - Extend user documentation with a warning about tools being Linux-only #3462
+ - Update cupla to latest dev #3212
```
Comment (psychocoderHPC) [inline CHANGELOG.md:319]: can be removed

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,6 +1,378 @@
...
+ - Update to latest cupla dev #3244
```
Comment (psychocoderHPC) [inline CHANGELOG.md:320]: can be removed

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,6 +1,378 @@
...
+ - Update hemera modules and add openPMDapi module #3270
+ - Provide gitlab CI yaml #3292
```
Comment (psychocoderHPC) [inline CHANGELOG.md:322]: can be removed

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,6 +1,378 @@
...
+ - Activate reduced HZDR CI tests #3282
+ - Set cmake version explicitly for travis CI #3305
+ - HZDR gitlab CI #3308
```
Comment (psychocoderHPC) [inline CHANGELOG.md:325]: can be removed

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,6 +1,378 @@
...
+ - Update to cupla 0.3.0-dev #3307
```
Comment (psychocoderHPC) [inline CHANGELOG.md:326]: can be removed

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,6 +1,378 @@
...
+ - Update openPMD and ADIOS module #3393
+ - Update summit profile to include openPMD #3384
+ - Add both adios and adios2 module on hemera #3411
+ - CI test with openpmd-api 0.12.0 #3422
+ - Add a new compile-time test suite to test all particle pushers #3386
+ - Use openPMD::getVersion() function #3427
+ - CI: check code style #3441
+ - CI: more test on HZDR CI #3443
+ - CI: perform branch check in HZDR CI #3449
+ - Update to latest cupla 0.3.0-dev #3454
```
Comment (psychocoderHPC) [inline CHANGELOG.md:340]: can be removed

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,6 +1,378 @@
...
+ - Delete superfluous `fieldSolver.param` in examples #3374
+ - Build CI tests with all dependencies #3316
+ - Update openPMD and ADIOS module #3393
+ - Update summit profile to include openPMD #3384
+ - Add both adios and adios2 module on hemera #3411
+ - CI test with openpmd-api 0.12.0 #3422
+ - Add a new compile-time test suite to test all particle pushers #3386
+ - Use openPMD::getVersion() function #3427
+ - CI: check code style #3441
+ - CI: more test on HZDR CI #3443
```
Comment (psychocoderHPC) [inline CHANGELOG.md:338]: can be removed

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,6 +1,378 @@
...
+ - CI: auto adjust parallelism #3450
+ - CI: skip compile #3452
+ - CI: new compiler and CUDA #3469
+ - Disable async device streams for CPU backends #3471
+ - CI: latest container v1.1 #3472
+ - Add SPEC benchmark example #3466
+ - Update to latest cupla with alpaka0.6.0rc4 #3474
+ - Remove openPMD 0.11.0 module for defq #3480
+ - Update summit profile #3492
+ - Update cupla #3501
```
Comment (psychocoderHPC) [inline CHANGELOG.md:350]: can be removed

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,6 +1,378 @@
...
+ - Update the FieldAbsorberTest example to match the Taflove book #3487
+ - Update CI #3509
+ - Update modules in JUWELS profiles #3459
+ - Update Hemera profiles to load the git module #3536
+ - Use HIP 4.0 in CI #3538
```
Comment (psychocoderHPC) [inline CHANGELOG.md:355]: can be removed

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,6 +1,378 @@
...
+ - Use openPMD::getVersion() function #3427
+ - CI: check code style #3441
+ - CI: more test on HZDR CI #3443
+ - CI: perform branch check in HZDR CI #3449
+ - Update to latest cupla 0.3.0-dev #3454
+ - CI: auto adjust parallelism #3450
+ - CI: skip compile #3452
+ - CI: new compiler and CUDA #3469
+ - Disable async device streams for CPU backends #3471
+ - CI: latest container v1.1 #3472
```
Comment (psychocoderHPC) [inline CHANGELOG.md:345]: can be removed

**Review by psychocoderHPC — CHANGES_REQUESTED** [review]
@sbastrakov Thanks for preparing the changelog. I marked a few PRs that can definitively be removed.

File: CHANGELOG.md (hunk 3 above)
Comment (psychocoderHPC) [reply CHANGELOG.md:17]: Agree with excluding or explicitly say "generating incident filed based  ... for Yee field solver".

**Comment by psychocoderHPC** [comment]
@sbastrakov  could you please finish the changelog that we can merge this pr tomorrow.

**Comment by psychocoderHPC** [comment]
@sbastrakov please add #3938 as bugfix to the changelog

File: CHANGELOG.md (hunk 5 above)
Comment (sbastrakov) [reply CHANGELOG.md:281]: I think only the last one is truly unimportant. The JUWELS profile could in principle be reused (rather copied-and-adjusted), and the other two are extension to docs. So i would slightly prefer keep the first 3

File: CHANGELOG.md (hunk 3 above)
Comment (sbastrakov) [reply CHANGELOG.md:17]: Clarified and shortened

**Comment by sbastrakov** [comment]
Log updated. I went again through recent backports and added the PRs to the log. Deleted almost all that you suggested, except some documentation PRs, see reply in review.

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**